evm/rlp/headers.py

```diff
@@ -133,3 +133,76 @@ def from_parent(cls,
 
         header = cls(**header_kwargs)
         return header
+
+
+class CollationHeader(rlp.Serializable):
+    fields = [
+        ("shard_id", big_endian_int),
+        ("expected_period_number", big_endian_int),
+        ("period_start_prevhash", hash32),
+        ("parent_hash", hash32),
+        ("coinbase", address),
+        ("transaction_root", hash32),
```

Comment:
The order of `coinbase` and `transaction_root` is wrong. It needs to follow spec for RLP objects.